xbmc/guilib/GUIPanelContainer.cpp

```diff
@@ -399,9 +399,17 @@ void CGUIPanelContainer::ValidateOffset()
 
 void CGUIPanelContainer::SetCursor(int cursor)
 {
+  // exceeds the number of items the panel can hold
   if (cursor > m_itemsPerPage * m_itemsPerRow - 1)
     cursor = m_itemsPerPage * m_itemsPerRow - 1;
-  if (cursor < 0) cursor = 0;
+
+  // exceeds the number of items being displayed
+  if (cursor > (int)m_items.size() - 1 - GetOffset() * m_itemsPerRow)
+    cursor = (int)m_items.size() - 1 - GetOffset() * m_itemsPerRow;
```

Comment:
Please, no c-style casts. Coding Guideline!

Reply:
You're right, thanks